### PR TITLE
Feat: Issue #32: Implement LIC 13

### LINKED ISSUE #32: Implement LIC 13 and add tests for it
Implement the Launch Interceptor Condition (LIC) number 13 in the `lic13` method which is defined in the `CMV` class.

The `lic13` method should return true if:

There exists at least one set of three data points, separated by exactly A PTS and B PTS
consecutive intervening points, respectively, that cannot be contained within or on a circle of
radius RADIUS1. In addition, there exists at least one set of three data points (which can be
the same or different from the three data points just mentioned) separated by exactly A PTS
and B PTS consecutive intervening points, respectively, that can be contained in or on a
circle of radius RADIUS2. Both parts must be true for the LIC to be true. The condition is
not met when NUMPOINTS < 5.
0 ≤ RADIUS2

Otherwise, the `lic13` method should return `false`.

When the `lic13` method is done, then add at least two tests for it in the `CMVTest` class:
- One test where the `lic13` method must return `true`.
- Another test where the `lic13` method must return `false`.

### PR DESCRIPTION
**Pull request by nattvara**
# About the PR

This PR will;

- Implement the LIC 13 conditions as specified by #32.
- Add two helper methods to the `CMVTest` test class (`createArrayWithPointsFarAwayFromEachOther` and `createArrayWithPointsAtOrigo`)
- Add a method `pointFitsInCircleWithRadius` to the `Point` class

Closes #32

### CODE REVIEW

File: src/test/java/decide/CMVTest.java [hunk 1]
```diff
@@ -316,9 +330,101 @@ void lic12Test() {
     }
 
     @Test
-    @DisplayName("LIC 13")
-    void lic13Test() {
-        assertTrue(true);
+    @DisplayName("LIC 13 Should be false if RADIUS2 is less than or equal to 2")
```
[the remark this answers is not in the document]
Comment (nattvara) [reply src/test/java/decide/CMVTest.java:333]: Did mean that, good catch!

File: src/main/java/decide/core/Point.java [hunk 2]
```diff
@@ -49,6 +49,20 @@ public double distance(Point to) {
...
+     *
+     * If the point p is on the circle it is considered within the circle.
+     *
+     * @param p         Point to check
+     * @param radius    Radius of the circle
+     * @return
+     */
+    public boolean pointFitsInCircleWithRadius(Point p, double radius) {
+        return distance(p) <= radius;
+    }
```
[the remark this answers is not in the document]
Comment (nattvara) [reply src/main/java/decide/core/Point.java:64]: Will do!

File: src/main/java/decide/core/CMV.java [hunk 3]
```diff
@@ -269,6 +269,39 @@ private boolean lic12() {
      * 0 ≤ RADIUS2
      */
     private boolean lic13() {
+        if (parameters.RADIUS2 <= 0) return false;
```
[the remark this answers is not in the document]
Comment (nattvara) [reply src/main/java/decide/core/CMV.java:331]: Doesn't hurt to keep these around

File: src/main/java/decide/core/CMV.java [hunk 4]
```diff
@@ -328,6 +328,34 @@ private boolean lic12() {
...
+        for (int i = 0; i < points.length - (parameters.A_PTS + 1) - (parameters.B_PTS + 1); i++) {
+            Point a = points[i];
+            Point b = points[i + parameters.A_PTS + 1];
+            Point c = points[i + (parameters.A_PTS + 1) + (parameters.B_PTS + 1)];
+
+            boolean withOrigoInA, withOrigoInB, withOrigoInC;
+
+            withOrigoInA = a.pointFitsInCircleWithRadius(b, parameters.RADIUS2) && a.pointFitsInCircleWithRadius(c, parameters.RADIUS2);
+            withOrigoInB = b.pointFitsInCircleWithRadius(a, parameters.RADIUS2) && b.pointFitsInCircleWithRadius(c, parameters.RADIUS2);
+            withOrigoInC = c.pointFitsInCircleWithRadius(a, parameters.RADIUS2) && c.pointFitsInCircleWithRadius(b, parameters.RADIUS2);
```
Comment (nattvara) [inline src/main/java/decide/core/CMV.java:353]: Don't understand MathTools.pointsAreCoveredByCircle well enough to write a MathTools.pointsAreNotCoveredByCircle, not as pretty keeping it like this, but don't feel it's worth the effort to redo it in a functional style

**Review by nattvara — COMMENTED** [review]
.

File: src/main/java/decide/core/CMV.java (hunk 4 above)
Comment (SamuSoft) [reply src/main/java/decide/core/CMV.java:353]: Isn't this just checking if the points are in range of each other? The maths for seeing if three points are covered by the same circle was quite a bit more complex than that. As long as you're using the pointsAreCoveredByCircle it'll be just one single call.

**Review by SamuSoft — APPROVED** [review]
Looks clean, ready to merge

**Pull request merged**